src/common/reactNativeProjectHelper.ts

```diff
@@ -19,22 +18,44 @@ export class ReactNativeProjectHelper {
      * {operation} - a function that performs the expected operation
      */
     public isReactNativeProject(): Q.Promise<boolean> {
-        let currentPackage = new Package(this.workspaceRoot);
-        return currentPackage.dependencies().then(dependencies => {
-            return !!(dependencies && dependencies["react-native"]);
-        }).catch((err: Error) => {
-            return Q.resolve(false);
+        let deferred = Q.defer<any>();
+
+        exec("react-native -v", {
+            cwd: this.workspaceRoot,
+        }, (error, stdout, stderr) => {
+            if (error) {
+                deferred.reject(error);
+            }
+
+            let version = stdout.toString().match(/react-native: ([\d\.]+)/);
+            deferred.resolve(version && semver.gte(version[1], "0.19.0"));
```

Comment:
The check for version 0.19.0 should only be in `validateReactNativeVersion()` (Where it was before). It doesn't make sense to check for a specific version in `isReactNativeProject()`. So yes, please remove it from here and just do the appropriate check to make sure this is a react-native project.